frontend/src/modules/Survey/Components/Survey.tsx

```diff
@@ -93,55 +104,61 @@ export const Survey = () => {
       ? courseList.length > 0 && courseList.every((c) => validCourseRe.test(c))
       : answers[multipleChoiceIndex] !== ''
 
-  return currStep === 1 ? ( // Form landing
-    <SplashBackground>
-      <StepBegin
-        name={nameAnswer}
-        email={emailAnswer}
-        setName={(arg: string) => setNameAnswer(arg)}
-        setEmail={(arg: string) => setEmailAnswer(arg)}
-        gotoNextStep={() => setCurrStep((currStep) => currStep + 1)}
-      />
-    </SplashBackground>
-  ) : currStep === totalSteps + 1 ? (
-    // Form confirmation
-    <QuestionBackground>
-      <StepFinal
-        success={surveyError === null}
-        submissionResponse={surveySubmissionResponse!}
-        errorMsg={surveyError != null ? surveyError : ''}
-      />
-    </QuestionBackground>
+  return currSurveyState ? (
```

Comment:
Can you also add a loading icon here? It can be an easy state where once the async retrieval of the status from the endpoint is done, you set a flag that you're done loading, and you show the page instead of a loader. 

The loader can be super simple too. See this: https://mui.com/material-ui/react-progress/